### ISSUE TITLE
[Web] The onnxruntime-web example is loading wasm file twice if set to local path

### ISSUE THREAD

**Issue by fquirin** [microsoft/onnxruntime#16028]
### Describe the issue

**Description:**

I'm running an Nginx web-server and placed the [quick-start example for onnxruntime-web with script-tag](https://github.com/microsoft/onnxruntime-inference-examples/tree/main/js/quick-start_onnxruntime-web-script-tag) in a folder together with the latest v1.14.0 `ort.js` and the `ort-wasm-simd.wasm` file. Then I changed the CDN path to localhost and when I start the example the `ort-wasm-simd.wasm` is simultaneously loaded twice.

**Expected:**

Load the `ort-wasm-simd.wasm` only once and wait until its finished.

**More info:**

A screenshot of the network tab in dev tools:

![image](https://github.com/microsoft/onnxruntime/assets/18550513/ebc25da9-edcd-4555-984e-312aac2f8a57)

A screenshot with the call stack for the 2nd wasm request:

![image](https://github.com/microsoft/onnxruntime/assets/18550513/4a2e2e94-2ef8-462e-8316-2de9cb29ac2b)

Warnings inside the dev console suggest that there is a MIME Type issue, but why should this trigger a full 2nd download?

![image](https://github.com/microsoft/onnxruntime/assets/18550513/5eed5b39-2b1a-43bb-930b-c2012d28c52d)

The HTML code:

```
<html>
    <header>
        <title>ONNX Runtime JavaScript examples: Quick Start - Web (using script tag)</title>
    </header>
    <body>
	<script src="ort.js"></script>
        <script>
            async function main() {
                try {
                    const session = await ort.InferenceSession.create('./model.onnx');
                } catch (e) {
                    document.write(`failed to inference ONNX model: ${e}.`);
                }
            }
            main();
        </script>
    </body>
</html>
```

### To reproduce

- Download the official quick-start example for onnxruntime-web with script-tag
- Download the `ort.js` and the `ort-wasm-simd.wasm` file from CDN
- Put the files in the same folder and adjust the script tag to load them locally
- Put everything on a local web-server
- Start the example and check the network tab to see what files are loaded

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

v1.14.0

### Execution Provider

'wasm'/'cpu' (WebAssembly CPU)

**Comment by guschmue** [comment]
true. I don't see it in my env because the server indicates "application/wasm". 
Same for https://cdn.jsdelivr.net/npm/onnxruntime-web/dist/.
Let me check if we can reuse the response if "application/wasm" wasn't send.
